python_modules/dagster/dagster/grpc/impl.py

```diff
@@ -399,3 +399,11 @@ def _error_message_fn(partition_set_name, partition_name):
         return ExternalPartitionExecutionErrorData(
             serializable_error_info_from_exc_info(sys.exc_info())
         )
+
+
+def get_notebook_data(notebook_path):
+    check.str_param(notebook_path, "notebook_path")
+
+    with open(os.path.abspath(notebook_path), "rb") as f:
```

Comment:
something about the arbitrary file access part here makes me nervous. Can we keep the    `if not path.endswith(".ipynb")` check from before here as well? Just imagining a weird attack scenario where somebody somehow uses this to get access to confidental files or something